lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,499 @@ module.exports = {
+                                                    return fixer.removeRange([getBeforeToken(node.parent), node.parent.range[1]]);
+                                                }
+
+                                                return fixer.removeRange([node.parent.range[0], getAfterToken(node.parent)]);
+                                            }
+
+                                            // remove unused varibles inside functions arguments
+                                            if (isTypeFunction(node.parent)) {
+                                                if (node.parent.params.length === 1) {
+                                                    return fixer.removeRange(node.range);
+                                                }
+
+                                                if (getTokenBeforeValue(node) === "(" && getTokenAfterValue(node) === ",") {
+                                                    return fixer.removeRange([node.range[0], getAfterToken(node)]);
+                                                }
+
+                                                return fixer.removeRange([getBeforeToken(node), node.range[1]]);
+                                            }
+
+                                            return null;
+                                        }
+
+                                        /**
+                                         * fix unused variables inside object and array pattern
+                                         * @param {ASTNode} node parent node to check
+                                         * @returns {Object} fixer object
+                                         */
+                                        function fixObjectAndArrayVariable(node) {
+                                            if (node.parent.type === "VariableDeclarator") {
+
+                                                // skip variable in for (const [ foo ] of bar);
+                                                if (isLoop(node.parent.parent.parent)) {
+                                                    return null;
+                                                }
+
+                                                if (node.parent.parent.declarations.length === 1) {
+                                                    return fixer.removeRange(node.parent.parent.range);
+                                                }
+
+                                                if (getTokenBeforeValue(node.parent) === ",") {
+                                                    return fixer.removeRange([getBeforeToken(node.parent), node.parent.range[1]]);
+                                                }
+
+                                                return fixer.removeRange([node.parent.range[0], getAfterToken(node.parent)]);
+                                            }
+
+                                            if (isTypeFunction(node.parent)) {
```

Comment:
Please add a comment describing what this does.

Also note, this is almost exactly the same as earlier in the file. Maybe there's a way to extract into a function?